### PR TITLE
[Merged by Bors] - feat: port Logic.Equiv.Fin

### PR DESCRIPTION
**Pull request by jcommelin**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by urkud** [comment]
I'm going to merge #1741 into this branch, then fix errors.

**Comment by jcommelin** [comment]
Oops, I messed up the rebase. Will push a fix asap.

File: Mathlib/Logic/Equiv/Fin.lean [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+#align prod_equiv_pi_fin_two prodEquivPiFinTwo
+
+/-- The space of functions `Fin 2 → α` is equivalent to `α × α`. See also `piFinTwoEquiv` and
+`prodEquivPiFinTwo`. -/
+@[simps (config := { fullyApplied := false })]
+def finTwoArrowEquiv (α : Type _) : (Fin 2 → α) ≃ α × α :=
+  { piFinTwoEquiv fun _ => α with invFun := fun x => ![x.1, x.2] }
+#align fin_two_arrow_equiv finTwoArrowEquiv
+
+/-- `Π i : Fin 2, α i` is order equivalent to `α 0 × α 1`. See also `order_iso.fin_two_arrow_equiv`
```
Comment (jcommelin) [inline Mathlib/Logic/Equiv/Fin.lean:94]:
```suggestion
/-- `Π i : Fin 2, α i` is order equivalent to `α 0 × α 1`. See also `OrderIso.finTwoArrowEquiv`
```

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**